File: test_api.js
const axios = require('axios');
async function test() {
    try {
        console.log("Registering user...", new Date());
        const timestamp = Date.now();
        const email = `test_dashboard_${timestamp}@test.com`;
        
        let res = await axios.post('http://127.0.0.1:8000/auth/register', {
            email: email,
            password: "TestPassword123!",
            full_name: "Tester",
            role: "tenant"
        });
        
        console.log("Logging in...", new Date());
        let loginFormData = new URLSearchParams();
        loginFormData.append('username', email);
        loginFormData.append('password', "TestPassword123!");
        
        res = await axios.post('http://127.0.0.1:8000/auth/login', loginFormData, {
            headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
        });
        const token = res.data.access_token;
        console.log("Token:", token.substring(0, 10) + "...");
        
        console.log("Fetching /auth/me ...", new Date());
        let meRes = await axios.get('http://127.0.0.1:8000/auth/me', {
            headers: { Authorization: `Bearer ${token}` }
        });
        console.log("Me status:", meRes.status);
        
        console.log("Fetching /auth/me/segment-config ...", new Date());
        let segmentRes = await axios.get('http://127.0.0.1:8000/auth/me/segment-config', {
            headers: { Authorization: `Bearer ${token}` }
        });
        console.log("Segment config status:", segmentRes.status);
        console.log("Done checking APIs!", new Date());

    } catch(err) {
        console.error("FAILED ERROR", err.response?.status, err.response?.data || err.message);
    }
}
test();

File: test_api.py
import requests
import datetime
import uuid

def test():
    try:
        email = f"test_dashboard_{uuid.uuid4().hex[:8]}@test.com"
        password = "TestPassword123!"
        
        print("Registering...", email)
        res = requests.post('http://127.0.0.1:8000/auth/register', json={
            "email": email,
            "password": password,
            "full_name": "Tester",
            "role": "tenant"
        })
        res.raise_for_status()
        
        print("Logging in...")
        res = requests.post('http://127.0.0.1:8000/auth/login', data={
            "username": email,
            "password": password
        })
        res.raise_for_status()
        token = res.json()["access_token"]
        
        headers = {"Authorization": f"Bearer {token}"}
        
        print("Fetching /auth/me ...")
        res = requests.get('http://127.0.0.1:8000/auth/me', headers=headers)
        res.raise_for_status()
        print("User data:", res.json()["email"])
        
        print("Fetching /auth/me/segment-config ...")
        res = requests.get('http://127.0.0.1:8000/auth/me/segment-config', headers=headers)
        res.raise_for_status()
        print("Segment Config:", res.json().get('segment'))
        
        print("Success!")
    except Exception as e:
        print("ERROR:", e)
        if hasattr(e, 'response') and e.response:
            print(e.response.text)

test()
